lib/azuremetadata/azuremetadata.py

```diff
@@ -133,3 +139,21 @@ def _make_request(url):
         print("An error occurred when fetching metadata:", file=sys.stderr)
         print(last_error, file=sys.stderr)
         return {}
+
+    @staticmethod
+    def _get_api(api_version):
+        if api_version == 'latest':
+            api_newest_versions = AzureMetadata._get_api_newest_versions()
+            api_version = api_newest_versions[0]
+        return api_version
+
+    @staticmethod
+    def _get_api_newest_versions():
+        newest_api = ['2017-04-02']
+        no_api_version_result = AzureMetadata._make_request(
```

Comment:
The naming here is rather confusing what follows is that we access `no_api_version_result` to return an API version but the variable name tells me there is no api version. Meaning the reader of this code has to be familiar with the missing Azure API for an API version query. So either a comment is in order that explains what's going on or a better variable naming. I think a comment is the preferable solution as we can be more elaborate.

Reply:
Good point. Fixed.